tests/e2e_test.go

```diff
@@ -0,0 +1,497 @@
+
+		// Send a transaction to the Teleporter contract
+		tx := newTestTeleporterTransaction(chainAIDInt, teleporterContractAddress, nonceA, data)
+
+		txSigner := types.LatestSignerForChainID(chainAIDInt)
+		signedTx, err := types.SignTx(tx, txSigner, fundedKey)
+		Expect(err).Should(BeNil())
+
+		subA, err := chainAWSClient.SubscribeNewHead(ctx, newHeadsA)
+		Expect(err).Should(BeNil())
+		defer subA.Unsubscribe()
+
+		log.Info("Sending Teleporter transaction on source chain", "destinationChainID", blockchainIDB, "txHash", signedTx.Hash())
+		err = chainARPCClient.SendTransaction(ctx, signedTx)
+		Expect(err).Should(BeNil())
+
+		// Sleep to ensure the new block is published to the subscriber
+		time.Sleep(5 * time.Second)
+		receipt, err := chainARPCClient.TransactionReceipt(ctx, signedTx.Hash())
+		Expect(err).Should(BeNil())
+		Expect(receipt.Status).Should(Equal(types.ReceiptStatusSuccessful))
+	})
+
+	ginkgo.It("Relay message to destination", ginkgo.Label("Teleporter", "RelayMessage"), func() {
+		ctx := context.Background()
+
+		// Get the latest block from Subnet A, and retrieve the warp message from the logs
+		log.Info("Waiting for new block confirmation")
+		newHeadA := <-newHeadsA
+		blockHashA := newHeadA.Hash()
+
+		log.Info("Fetching relevant warp logs from the newly produced block")
+		logs, err := chainARPCClient.FilterLogs(ctx, interfaces.FilterQuery{
+			BlockHash: &blockHashA,
+			Addresses: []common.Address{warp.Module.Address},
+		})
+		Expect(err).Should(BeNil())
+		Expect(len(logs)).Should(Equal(1))
+
+		// Check for relevant warp log from subscription and ensure that it matches
+		// the log extracted from the last block.
+		txLog := logs[0]
+		log.Info("Parsing logData as unsigned warp message")
+		unsignedMsg, err := avalancheWarp.ParseUnsignedMessage(txLog.Data)
+		Expect(err).Should(BeNil())
+
+		// Set local variables for the duration of the test
+		unsignedWarpMessageID := unsignedMsg.ID()
+		unsignedWarpMsg := unsignedMsg
+		log.Info("Parsed unsignedWarpMsg", "unsignedWarpMessageID", unsignedWarpMessageID, "unsignedWarpMessage", unsignedWarpMsg)
+
+		// Loop over each client on chain A to ensure they all have time to accept the block.
+		// Note: if we did not confirm this here, the next stage could be racy since it assumes every node
+		// has accepted the block.
+		for i, uri := range chainANodeURIs {
+			chainAWSURI := httpToWebsocketURI(uri, blockchainIDA.String())
+			log.Info("Creating ethclient for blockchainA", "wsURI", chainAWSURI)
+			client, err := ethclient.Dial(chainAWSURI)
+			Expect(err).Should(BeNil())
+
+			// Loop until each node has advanced to >= the height of the block that emitted the warp log
+			for {
+				block, err := client.BlockByNumber(ctx, nil)
+				Expect(err).Should(BeNil())
+				if block.NumberU64() >= newHeadA.Number.Uint64() {
+					log.Info("client accepted the block containing SendWarpMessage", "client", i, "height", block.NumberU64())
+					break
+				}
+			}
+		}
```

Comment:
Would it be possible to somehow move this to a `waitForAllValidatorsToAcceptBlock` helper? I imagine it could be useful for other tests in the future.

Reply:
Added a helper